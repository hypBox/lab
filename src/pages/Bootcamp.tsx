import React, { useState, useEffect } from "react"

import { Grid, Button } from "../components/material/core"
import { makeStyles } from "../components/material/styles"
import ProgressSidebar from "../components/bootcamp/progressSidebar"
import { bootcampData } from "../api"
import { ITask, IMilestone } from "../types"
import Task from "../components/bootcamp/task"

const useStyles = makeStyles((theme) => ({
  main: {},
  progress: {
    backgroundColor: "green",
  },
  content: {
    backgroundColor: "yellow",
  },
}))

const Bootcamp = () => {
  const classes = useStyles()
  const [showingPrgbar, setShowingPrgbar] = useState(true)
  const getSize = (isShowing: boolean): any =>
    isShowing
      ? { xs: 12, sm: 8, md: 9, lg: 10, xl: 11 }
      : { xs: 12, sm: 12, md: 12, lg: 12, xl: 12 }
  const [size, setSize] = useState(getSize(showingPrgbar))

  const [selectedTask, setTask] = useState<ITask>(
    bootcampData.milestones[0].tasks[0]
  )
  const [selectedMilestone, setMilestone] = useState<IMilestone>(
    bootcampData.milestones[0]
  )

  const milestoneClickHandler = (milestone: IMilestone) => {
    setMilestone(milestone)
  }

  const taskClickHandler = (task: ITask) => {
    setTask(task)
  }

  useEffect(() => {
    console.log("selectedTask", selectedTask) // handle change
  }, [selectedTask])

  useEffect(() => {
    setSize(getSize(showingPrgbar))
  }, [showingPrgbar])

  const getText = () => (showingPrgbar ? "Hide" : "Show")

  const clickHandler = () => {
    setShowingPrgbar(!showingPrgbar)
  }

  return (
    <Grid container className={classes.main} direction="row">
      <Grid {...size} item className={classes.content}>
        <Task task={selectedTask} />
        <Button variant="contained" onClick={clickHandler} color="primary">
          {getText()}
        </Button>
        {selectedMilestone.title} >> {selectedTask.title}
      </Grid>

      {showingPrgbar && (
        <Grid
          data-testid="progress-sidebar"
          item
          xs={12}
          sm={4}
          md={3}
          lg={2}
          xl={1}
          className={classes.progress}
        >
          <ProgressSidebar
            onMilestoneClick={milestoneClickHandler}
            onTaskClick={taskClickHandler}
            selectedMilestone={selectedMilestone}
            selectedTask={selectedTask}
            bootcamp={bootcampData}
          />
        </Grid>
      )}
    </Grid>
  )
}

export default Bootcamp
