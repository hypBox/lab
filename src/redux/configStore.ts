import { createStore, Store, applyMiddleware } from "redux"
import rootReducer from "./rootReducer"

import { createLogger } from "redux-logger"
import { composeWithDevTools } from "redux-devtools-extension"
import reduxImmutableStateInvariant from "redux-immutable-state-invariant"

export interface AppState {
  preferences: {
    darkMode: boolean
  }
}

const middlewares: Array<any> = [reduxImmutableStateInvariant()]

if (process.env.NODE_ENV === "development") {
  middlewares.push(
    createLogger({
      collapsed: true,
    })
  )
}

const configStore = (): Store<AppState> =>
  createStore(rootReducer, composeWithDevTools(applyMiddleware(...middlewares)))

export default configStore
