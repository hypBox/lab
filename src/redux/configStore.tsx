import { createStore } from "redux"
import rootReducer from "./reducers"

export interface Store {
  darkTheme: boolean
}

const configStore = (intialState: any) => createStore(rootReducer, intialState)

export default configStore
