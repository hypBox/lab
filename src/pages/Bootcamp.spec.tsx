import React from "react"
import { render, fireEvent } from "@testing-library/react"
import { Provider as ReduxProvider } from "react-redux"
import configStore from "../redux/configStore"

import Bootcamp from "./Bootcamp"

describe("Bootcamp", () => {
  const renderBootcampPage = () => {
    const store = configStore()
    return render(
      <ReduxProvider store={store}>
        <Bootcamp />
      </ReduxProvider>
    )
  }

  it("renders as expected", () => {
    const tree = renderBootcampPage()
    expect(tree).toMatchSnapshot()
  })

  it("verifies progressSidebar is shown", () => {
    const tree = renderBootcampPage()
    const sidebar = tree.findByTestId("progress-sidebar")
    expect(sidebar).toBeTruthy()
  })

  it("hides progressSidebar", async (done) => {
    const { queryByTestId, getByRole } = renderBootcampPage()
    const button = getByRole("toggle-progress-sidebar")
    fireEvent.click(button)
    const progressSidebar = await queryByTestId("progress-sidebar")
    expect(progressSidebar).toBeFalsy()
    done()
  })
})
