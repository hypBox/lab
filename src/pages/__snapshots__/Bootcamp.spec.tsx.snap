// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bootcamp renders as expected 1`] = `
<div
  className="MuiGrid-root makeStyles-main-1 MuiGrid-container"
>
  <div
    className="MuiGrid-root makeStyles-content-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-9 MuiGrid-grid-lg-10"
  >
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-appBar-107 MuiPaper-elevation4"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          className="MuiTypography-root makeStyles-title-106 MuiTypography-h6"
        >
          Bootcamp Overview
        </h6>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-moon fa-w-16 fa-lg "
              data-icon="moon"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M283.211 512c78.962 0 151.079-35.925 198.857-94.792 7.068-8.708-.639-21.43-11.562-19.35-124.203 23.654-238.262-71.576-238.262-196.954 0-72.222 38.662-138.635 101.498-174.394 9.686-5.512 7.25-20.197-3.756-22.23A258.156 258.156 0 0 0 283.211 0c-141.309 0-256 114.511-256 256 0 141.309 114.511 256 256 256z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="toggle-progress-sidebar"
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-bars fa-w-14 fa-lg "
              data-icon="bars"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
        </button>
      </div>
    </header>
    <div
      className="makeStyles-playerWrapper-213"
    >
      <div
        className="react-player"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-3 MuiGrid-grid-lg-2"
    data-testid="progress-sidebar"
  >
    <nav
      aria-labelledby="nested-list-subheader"
      className="MuiList-root makeStyles-root-215 MuiList-padding"
    >
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-check fa-w-16 "
            data-icon="check"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-displayBlock"
          >
            Overview
          </span>
        </div>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-chevron-down fa-w-14 makeStyles-milestoneIcon-220"
          data-icon="chevron-down"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <div
        className="MuiCollapse-container MuiCollapse-entered"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper"
        >
          <div
            className="MuiCollapse-wrapperInner"
          >
            <div
              className="MuiList-root MuiList-dense"
            >
              <li
                className="MuiListItem-container"
              >
                <div
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root makeStyles-nested-245 MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction Mui-selected"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <div
                    className="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-check fa-w-16 "
                      data-icon="check"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </div>
                  <div
                    className="MuiListItemText-root MuiListItemText-dense"
                  >
                    <span
                      className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-displayBlock"
                    >
                      Bootcamp Overview
                    </span>
                  </div>
                </div>
                <div
                  className="MuiListItemSecondaryAction-root"
                >
                  <span
                    className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextPrimary"
                  >
                    1m 24s
                  </span>
                </div>
              </li>
              <li
                className="MuiListItem-container"
              >
                <div
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root makeStyles-nested-245 MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <div
                    className="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-check fa-w-16 "
                      data-icon="check"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </div>
                  <div
                    className="MuiListItemText-root MuiListItemText-dense"
                  >
                    <span
                      className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-displayBlock"
                    >
                      Course Outline
                    </span>
                  </div>
                </div>
                <div
                  className="MuiListItemSecondaryAction-root"
                >
                  <span
                    className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextPrimary"
                  >
                    2m 22s
                  </span>
                </div>
              </li>
              <li
                className="MuiListItem-container"
              >
                <div
                  aria-disabled={false}
                  className="MuiButtonBase-root MuiListItem-root makeStyles-nested-245 MuiListItem-dense MuiListItem-gutters MuiListItem-button MuiListItem-secondaryAction"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onDragLeave={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  role="button"
                  tabIndex={0}
                >
                  <div
                    className="MuiListItemIcon-root"
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-check fa-w-16 "
                      data-icon="check"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </div>
                  <div
                    className="MuiListItemText-root MuiListItemText-dense"
                  >
                    <span
                      className="MuiTypography-root MuiListItemText-primary MuiTypography-body2 MuiTypography-colorTextPrimary MuiTypography-displayBlock"
                    >
                      Project Overview
                    </span>
                  </div>
                </div>
                <div
                  className="MuiListItemSecondaryAction-root"
                >
                  <span
                    className="MuiTypography-root MuiTypography-caption MuiTypography-colorTextPrimary"
                  >
                    4m 32s
                  </span>
                </div>
              </li>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-disabled={false}
        className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiListItemIcon-root"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-layer-group fa-w-16 "
            data-icon="layer-group"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.41 148.02l232.94 105.67c6.8 3.09 14.49 3.09 21.29 0l232.94-105.67c16.55-7.51 16.55-32.52 0-40.03L266.65 2.31a25.607 25.607 0 0 0-21.29 0L12.41 107.98c-16.55 7.51-16.55 32.53 0 40.04zm487.18 88.28l-58.09-26.33-161.64 73.27c-7.56 3.43-15.59 5.17-23.86 5.17s-16.29-1.74-23.86-5.17L70.51 209.97l-58.1 26.33c-16.55 7.5-16.55 32.5 0 40l232.94 105.59c6.8 3.08 14.49 3.08 21.29 0L499.59 276.3c16.55-7.5 16.55-32.5 0-40zm0 127.8l-57.87-26.23-161.86 73.37c-7.56 3.43-15.59 5.17-23.86 5.17s-16.29-1.74-23.86-5.17L70.29 337.87 12.41 364.1c-16.55 7.5-16.55 32.5 0 40l232.94 105.59c6.8 3.08 14.49 3.08 21.29 0L499.59 404.1c16.55-7.5 16.55-32.5 0-40z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </div>
        <div
          className="MuiListItemText-root"
        >
          <span
            className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-colorTextPrimary MuiTypography-displayBlock"
          >
            Variables and Types
          </span>
        </div>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-chevron-up fa-w-14 makeStyles-milestoneIcon-220"
          data-icon="chevron-up"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
    </nav>
  </div>
</div>
`;
